File: LeetCodeProblemsJava/src/Problems/LeetCode3300MinimumElementAfterReplacementWithDigitSum.java
package Problems;

public class LeetCode3300MinimumElementAfterReplacementWithDigitSum {
    public int minElement(int[] nums) {
        int minNum= 1<<30;
        for (var num : nums) {
            minNum=Math.min(minNum,getDigitSum(num));
        }
        return minNum;
    }
    int getDigitSum(int num) {
        int sum=0;
        while (num!=0) {
            sum+=num%10;
            num/=10;
        }
        return sum;
    }
}

File: LeetCodeProblemsGo/LeetCode3300MinimumElementAfterReplacementWithDigitSum/LeetCode3300MinimumElementAfterReplacementWithDigitSum.go
package LeetCode3300MinimumElementAfterReplacementWithDigitSum

import "math"

func minElement(nums []int) int {
	minDigitSum := math.MaxInt32
	for _, num := range nums {
		minDigitSum = min(minDigitSum, getDigitSum(num))
	}
	return minDigitSum
}
func getDigitSum(num int) int {
	sum := 0
	for num > 0 {
		sum += num % 10
		num /= 10
	}
	return sum
}
